core/trino-main/src/main/java/io/trino/execution/DynamicFiltersCollector.java

```diff
@@ -88,6 +120,14 @@ public synchronized void acknowledge(long callersCurrentVersion)
         dynamicFilterDomains.values().removeIf(domain -> domain.getVersion() <= callersCurrentVersion);
```

Comment:
In practise for every DynamicFilterId we will receive only the final domain once. In the future, it is theoretically possible that each dynamic filter's domain is being narrowed over time. Even in that case I think just replacing old domain with new one would work instead of intersect. I don't think we support incremental collection of domain here, every version of the collected domain should be either the same or a subset of the previous.
So the intersect looks confusing to me as well, maybe @sopel39 why it was written like that. It definitely doesn't have any impact currently as domains are delivered in single shot.